package/MDAnalysis/analysis/hydrogenbonds/hbond_analysis.py

```diff
@@ -316,10 +316,20 @@ def __init__(self, universe,
 
         self.u = universe
         self._trajectory = self.u.trajectory
+
         self.donors_sel = donors_sel
         self.hydrogens_sel = hydrogens_sel
         self.acceptors_sel = acceptors_sel
 
+        msg = ("{} is an empty selection string - no hydrogen bonds will "
+               "be found. This may be intended, but please check your "
+               "selection."
+               )
+        for sel in ['donors_sel', 'hydrogens_sel', 'acceptors_sel']:
+            val = getattr(self, sel)
+            if isinstance(val, str) and not val:
```

Comment:
I think my previous comment may have been lost in the discussions - I think you should be using `val.strip()` here to also catch the edge case of someone passing `" "` (which will evaluate to `True`)

Reply:
Ah yes, sorry about that! I'm pretty busy now (I have my viva tomorrow), but I'll sort it out either at the weekend or early next week

Reply:
Since it's not a large change, if you don't mind I'll just push a quick update to deal with this w/ associated test. That way I can push this in to the 2.1.0 release.

Reply:
That's would be great if you could take care of it.

Thank you!

Reply:
Thanks for merging the latest develop into this branch.


Rather than doing `sel.strip()` here, I've stripped the strings of whitespace when assigning them as instance attributes. This is because `select_atoms` [doesn't strip the whitespace](https://github.com/MDAnalysis/mdanalysis/blob/develop/package/MDAnalysis/core/groups.py#L3133-L3138), and so passing a whitespace-only string to it raises a `SelectionError`:

`Unknown selection token: 'None'`.

Reply:
There's possibly a viewpoint where getting the `SelectionError` might be a better thing, but I'm not sure it's worth holding up this PR.

Reply:
Yeah probably not worth holding up the PR, but IMO whitespace etc should be addressed in the selection parser itself -- raised #3487 for this.

Reply:
Thanks for raising the issue @lilyminium !